pkg/auth/rbac.go

```diff
@@ -109,18 +131,22 @@ func (r *Rbac) Run(ctx context.Context) {
 				continue
 			}
 			msgType := binary.LittleEndian.Uint32(rec.RawSample)
+			tupleData := rec.RawSample[unsafe.Sizeof(msgType):]
 			var buf *bytes.Buffer
 			switch msgType {
 			case MSG_TYPE_IPV4:
-				buf = bytes.NewBuffer(rec.RawSample[4 : IPV4_TUPLE_LENGTH+4+1])
-				if err = binary.Read(buf, binary.LittleEndian, &tupleV4); err != nil {
+				// The valid length of IPv4 tuple is IPV4_TUPLE_LENGTH
+				buf = bytes.NewBuffer(tupleData[:IPV4_TUPLE_LENGTH+1])
+				// Read as big endian, then buf is host order
+				if err = binary.Read(buf, binary.BigEndian, &tupleV4); err != nil {
```

Comment:
why change to BigEndian, isn't local port storred in host byte order

Reply:
`buf` is read AS big-endian, thus it is host order after `binary.Read` is called